bundles/org.openhab.binding.airq/src/main/java/org/openhab/binding/airq/internal/airqHandler.java

```diff
@@ -0,0 +1,986 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+/**
+ * The {@link airqHandler} is responsible for retrieving all information from the air-Q device
+ * and change properties and channels accordingly.
+ *
+ * @author Aurelio Caliaro - Initial contribution
+ */
+@NonNullByDefault
+public class airqHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(airqHandler.class);
+    private @Nullable ScheduledFuture<?> pollingJob;
+    private @Nullable ScheduledFuture<?> getConfigDataJob;
+    private String ipaddress;
+    private String password;
+    private @Nullable ThingStatus thStatus;
+    protected static final int POLLING_PERIOD_DATA = 15000; // in milliseconds
+    protected static final int POLLING_PERIOD_CONFIG = 1; // in minutes
+
+    final class ResultPair {
+        private final float value;
+        private final float maxdev;
+
+        public float getvalue() {
+            return value;
+        }
+
+        public float getmaxdev() {
+            return maxdev;
+        }
+
+        // ResultPair() expects a string formed as this: [1234,56,789,012] and gives back a ResultPair
+        // consisting of the two numbers
+        public ResultPair(String input) {
+            value = Float.parseFloat(input.substring(1, input.indexOf(',')));
+            maxdev = Float.parseFloat(input.substring(input.indexOf(',') + 1, input.length() - 1));
+        }
+    }
+
+    public airqHandler(Thing thing) {
+        super(thing);
+        ipaddress = "";
+        password = "";
+    }
+
+    public Boolean ohCmd2airqCmd(String ohcmd) {
+        switch (ohcmd) {
+            case "ON":
+                return true;
+            case "OFF":
+                return false;
+        }
+        return false;
+    }
+
+    private boolean isTimeFormat(String str) {
+        try {
+            LocalTime.parse(str);
+        } catch (DateTimeParseException e) {
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.info(
+                "air-Q - airqHandler - handleCommand(): request received to handle value {} command {} of channelUID={}",
+                command, command.getClass(), channelUID);
+        if ((command instanceof OnOffType) || (command instanceof StringType)) {
+            JsonObject newobj = new JsonObject();
+            JsonObject subjson = new JsonObject();
+            switch (channelUID.getId()) {
+                case "wifi":
+                    // we do not allow to switch off Wifi because otherwise we can't connect to the air-Q device anymore
+                    break;
+                case "wifiInfo":
+                    newobj.addProperty("WifiInfo", ohCmd2airqCmd(command.toString()));
```

Comment:
```suggestion
                    newobj.addProperty("WifiInfo", command == OnOffType.ON);
```
If the above suggestion works you eliminate 1. a conversion to a string and then 2. comparing two strings which is more costly then doing this. Same comment to all the times you use the ohCmd2airqCmd() function.